hbase-common/src/main/java/org/apache/hadoop/hbase/PrivateCellUtil.java

```diff
@@ -768,11 +768,13 @@ public static boolean qualifierStartsWith(final Cell left, final byte[] startsWi
       throw new IllegalArgumentException("Cannot pass an empty startsWith");
     }
     if (left instanceof ByteBufferExtendedCell) {
-      return ByteBufferUtils.equals(((ByteBufferExtendedCell) left).getQualifierByteBuffer(),
+      return ((ByteBufferExtendedCell) left).getQualifierLength() >= startsWith.length
```

Comment:
nit: cast is redundant here. `left.getQualifierLength() >= startsWith.length` should work

Reply:
thanks.